packages/block-library/src/paragraph/edit.js

```diff
@@ -103,8 +103,16 @@ class ParagraphBlock extends Component {
 
 		if ( after !== null ) {
 			// Append "After" content as a new paragraph block to the end of
-			// any other blocks being inserted after the current paragraph.
-			blocks.push( createBlock( name, { content: after } ) );
+			// any other blocks being inserted after the current paragraph
+			if ( after.length ) {
+				// If after has content, keep the new block styling same as current one.
+				const newBlockAttributes = attributes;
+				newBlockAttributes.content = after;
```

Comment:
Ideally we don't mutate the attributes object of the current object, this could be replaced by

```
const newBlockAttributes = {
   ...attributes,
   content: after
};
```